UnityProject/Assets/Scripts/HealthV2/Living/Cyborg/BatteryPack.cs

```diff
@@ -1,36 +1,79 @@
-using System;
-using System.Collections;
 using System.Collections.Generic;
+using Systems.Construction.Parts;
 using UnityEngine;
 
-public class BatteryPack : MonoBehaviour
+namespace HealthV2.Living.Cyborg
 {
-	public List<Battery> Cells = new List<Battery>();
+	public class BatteryPack : MonoBehaviour, IChargeable
+	{
+		public List<Battery> Cells = new List<Battery>();
 
-	public ItemStorage Storage;
+		public ItemStorage Storage;
 
+		public bool IsFullyCharged
+		{
+			get
+			{
+				bool fullyCharged = true;
+				foreach (var battery in Cells)
+				{
+					if (battery.IsFullyCharged == false)
+					{
+						fullyCharged = false;
+						break;
+					}
+				}
 
-	public void Awake()
-	{
-		Storage.ServerInventoryItemSlotSet += BodyPartTransfer;
-	}
+				return fullyCharged;
+			}
+		}
 
-	private void BodyPartTransfer(Pickupable prevImplant, Pickupable newImplant)
-	{
-		Battery Battery = null;
-		if (newImplant && newImplant.TryGetComponent(out Battery))
+
+
+		public void ChargeBy(float watts)
 		{
-			if (Cells.Contains(Battery) == false)
+			Battery toCharge = null;
+
+			//Code that charges Each battery individually until they're all full
```

Comment:
am I just stoopid? I don't see how this code will charge each battery individually when the part that charges is outside the foreach loop